### PR TITLE
Validate OpenIdServerSettings before registering OpenId server

### PR DESCRIPTION
**Pull request by MatthijsKrempel**
Validated OpenIdServerSettings before starting the OpenId server, so that it does not throw exceptions.

Described in issue https://github.com/OrchardCMS/OrchardCore/issues/2414

### CODE REVIEW

**Comment by MatthijsKrempel** [comment]
@PinpointTownes it's not pretty, but at least it is no longer throwing exceptions

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs [hunk 1]
```diff
@@ -411,6 +411,8 @@ private async Task<IActionResult> ExchangeAuthorizationCodeOrRefreshTokenGrantTy
             var principal = await _signInManager.CreateUserPrincipalAsync(user);
             var identity = (ClaimsIdentity) principal.Identity;
 
+            identity.AddClaim(new Claim("TrivestId", "Contoso"));
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs]: ?

File: src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs [hunk 2]
```diff
@@ -116,69 +117,80 @@ public override void ConfigureServices(IServiceCollection services)
...
             {
-                var openIddictServerOptions = serviceProvider
-                    .GetRequiredService<IOptionsMonitor<OpenIddictServerOptions>>()
-                    .Get(OpenIddictServerDefaults.AuthenticationScheme);
+                var validationResult = servic.ValidateSettingsAsync(settings)
+                    .GetAwaiter()
+                    .GetResult();
 
-                if (openIddictServerOptions.AuthorizationEndpointPath != PathString.Empty)
+                if (validationResult.Length == 0)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs]: I would have thought that only this line would be sufficient to fix the issue.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs (hunk 1 above)
Comment (MatthijsKrempel) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs]: This is not supposed to be in there, going to remove it right away.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs (hunk 2 above)
Comment (MatthijsKrempel) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs]: The admin view also uses the same mechanic, so I had to rewrite that.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdServerConfiguration.cs [hunk 3]
```diff
@@ -123,6 +123,11 @@ public void Configure(string name, OpenIddictServerOptions options)
                 options.AccessTokenHandler = new JwtSecurityTokenHandler();
             }
 
+            if (!settings.EnableAuthorizationEndpoint && !settings.EnableTokenEndpoint && !settings.EnableLogoutEndpoint && !settings.EnableUserInfoEndpoint)
+            {
+                // Work around for Openiddict that throws an exception when no flows are selected
+                settings.EnableAuthorizationEndpoint = true;
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdServerConfiguration.cs]: Yeah no, we can't do that (enabling disabled endpoints is just evil).
I'm not even sure it fully fixes the issue as you'll probably get an exception indicating no flow was configured, even if you enable the authorization endpoint.

**Comment by PinpointTownes** [comment]
See https://github.com/OrchardCMS/OrchardCore/pull/2487 for a different approach using `OpenIdServerSettings` as the direct configuration source for the route paths.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdServerConfiguration.cs (hunk 3 above)
Comment (MatthijsKrempel) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdServerConfiguration.cs]: Going to remove this, since this is no longer neccesairy

**Comment by MatthijsKrempel** [comment]
Will review, if it fixes these changes, I'll delete this PR.

**Pull request closed without merging**